### ISSUE TITLE
Investigate MicroProfile OpenAPI 1.1

### ISSUE THREAD

**Issue by jmini** [smallrye/smallrye-open-api#44]
With the release of [microprofile-open-api](https://github.com/eclipse/microprofile-open-api) planned mid December 2018, I have started to experiment with `1.1-SNAPSHOT` of the API.

You can find the code in the [1.1](https://github.com/jmini/smallrye-open-api/tree/1.1) branch of my fork.
I am not sure it meet all your standard and requirements, but at least it compiles (and the tests are green). 

If you are interested to have this somewhere in this repository, let me know.
Otherwise I can wait until version `1.1` of the API is released and then propose a PR against `master`

**Comment by EricWittmann** [comment]
Hey this is great stuff, thanks!  I was planning on waiting for 1.1 of the spec to be released and then spend a week or so updating SmallRye OpenAPI, so this saves me a ton of work.  Really appreciate the contribution.  Let's go ahead and create a 1.1 branch for SmallRye so we can iterate and collaborate on this stuff.

I'll get that started on Monday...

**Comment by msavy** [comment]
I also had a look at this earlier today and think it looks good — only bit that stood out is the method that’s pending removal, but that should be cleared up soon by upstream. 👍

**Comment by EricWittmann** [comment]
Yeah sorry, by "Let's go ahead" I was referring to myself. :)  I'll create a 1.1 branch tomorrow AM.

**Comment by jmini** [comment]
I have noticed a problem: `ModelConstructionTckTest` is green because none of the tests are run.
I guess this is a TestNG / JUnit issue...

Smallrye is running JUnit Surefire and the TCK Tests are written with TestNG (`@org.testng.annotations.Test`). The tests are ignored.

Have you an idea how to solve this?

---
When this test run, there will be some more work to do.

One thing that was changed with `1.1` and that should fail here: `set/getDefaultValue()` in `APIResponses` is just a convenience for the `APIResponses.DEFAULT` key.

**Comment by EricWittmann** [comment]
In your 1.1 changes you've changed the `ModelContstructionTckTest` so that it no longer extends `BaseTckTest`.  I think this link should work to the relevant portion of the diff:  

https://github.com/smallrye/smallrye-open-api/compare/1.1...jmini:1.1#diff-bda46ad94feb21daec9ce72eb976e373

Try reverting that change.  I was wondering about that change....

**Comment by EricWittmann** [comment]
Also - I have created a `1.1` branch in SmallRye now.

**Comment by jmini** [comment]
> In your 1.1 changes you've changed the `ModelContstructionTckTest` so that it no longer extends `BaseTckTest`.

Even if the diff is not working for me (probably it will be better once there is PR), the problem is that `ModelContstructionTest` in the official TCK is no longer an Arquillian test.
See the discussion here: https://github.com/eclipse/microprofile-open-api/pull/274

When I proposed this change in the TCK, I was wondering if this would be a problem for vendors...

On the other side, requiring a server (with `@Deployment public static WebArchive createDeployment()`) for tests that only manipulates interfaces for simple beans is an overkill (and not working for me). 

As I have discussed in the PR, if you think this is the solution, we can restore `extends Arquillian` in the TCK (as long as we do not have the `createDeployment()` method, it is OK for my use case)
